src/mongo_client.ts

```diff
@@ -380,6 +404,15 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
       sessionPool: new ServerSessionPool(this),
       activeSessions: new Set(),
 
+      getAuthProvider(name: AuthMechanism | string) {
+        if (!clientAuthProviders[name]) {
+          const provider = AUTH_PROVIDERS.get(name)?.();
+          if (provider) {
+            clientAuthProviders[name] = provider;
+          }
```

Comment:
I think we should probably throw here if there's no provider to allude that an invalid name was passed in.